### ISSUE TITLE
Question about performance regression (master vs v1.0.1)

### ISSUE THREAD

**Issue by HyukjinKwon** [jimhester/lintr#270]
I lately tried to upgrade from jimhester/lintr@a769c0b to https://github.com/jimhester/lintr/commit/5431140ffea65071f1327625d4a8de9688fa7e72. The master branch, https://github.com/jimhester/lintr/commit/5431140ffea65071f1327625d4a8de9688fa7e72 (vs  jimhester/lintr@a769c0b) looks it has fixed many bugs and now finds many instances that I have observed and thought should be caught time to time in R code base in a project.

The problem is, it now takes about 7ish mins, (it was 2ish mins before) in my local. It is not the problem with v1.0.1 (it also takes 2ish mins).

To verify this, I filtered out new litners in the master branch, https://github.com/jimhester/lintr/commit/5431140ffea65071f1327625d4a8de9688fa7e72 (vs  v1.0.1):

Assuming from the doc, v1.0.1 looks missing the linters below (vs master, https://github.com/jimhester/lintr/commit/5431140ffea65071f1327625d4a8de9688fa7e72):

```
nonportable_path_linter: check that file.path() is used to construct safe and portable paths.
pipe_continuation_linter: Check that each step in a pipeline is on a new line, or the entire pipe fits on one line.
extraction_operator_linter: check that the [[ operator is used when extracting a single element from an object, not [ (subsetting) nor $ (interactive use).
implicit_integer_linter: check that integers are explicitly typed using the form 1L instead of 1.
object_name_linter: check that object names conform to a single naming style, e.g. snake_case or lowerCamelCase.
semicolon_terminator_linter: check that no semicolons terminate statements.
todo_comment_linter: check that the source contains no TODO comments (case-insensitive).
T_and_F_symbol_linter: avoid the symbols T and F (for TRUE and FALSE).
undesirable_function_linter: report the use of undesirable functions, e.g. options or sapply and suggest an alternative.
undesirable_operator_linter: report the use of undesirable operators, e.g. ::: or <<- and suggest an alternative.
unneeded_concatenation_linter: check that the c function is not used without arguments nor with a single constant
```

I manually disabled the linters above but the elapsed time looks almost the same, 7ish mins. So, it looks existing linters have some problem with performance regression.

I wonder how much it helps but I tried to disable everything except for each linter [here](https://gist.github.com/HyukjinKwon/1f852bcdcc2a13f396dfd2d6a88bea16) with the code base here, https://github.com/apache/spark/tree/master/R

To cut it short, are there any advices to avoid the slowdown between master branch and v1.0.1, or dose anyone know roughly why?

I would like to use the master one as it detects many meaningful instances but the slowdown is a concern.

**Comment by HyukjinKwon** [comment]
I wonder if it helps but what I mean by "meaningful instances" is, [here](https://gist.github.com/HyukjinKwon/4f59ddcc7b6487a02da81800baca533c). Most of them were related with the length (I guess `line_length_linter`) and and no spaces check (I guess `spaces_left_parentheses_linter`).

**Comment by jimhester** [comment]
Some issues were discussed in https://github.com/jimhester/lintr/issues/226, and this is mostly solved I believe.
